cmd/inspectImages.go

```diff
@@ -0,0 +1,143 @@
+// Copyright © 2018 openSUSE opensuse-project@opensuse.org
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/openSUSE/helm-mirror/formatter"
+	"github.com/openSUSE/helm-mirror/service"
+	"github.com/openSUSE/helm-mirror/utils"
+
+	"github.com/spf13/cobra"
+)
+
+var imagesFile string
+var output string
+var target string
+
+const imagesDesc = `Extract all the images of the Helm Chart or
+the Helm Charts in the folder provided. This command dumps
+the images on 'stdout' by default, for more options check
+'output' flag. Example:
+
+  - helm-mirror inspect-images /tmp/helm
+
+  - helm-mirror inspect-images /tmp/helm/app.tgz
+
+The [folder|tgzfile] has to be a full path.
+`
+
+const outputDesc = `choose an output for the list of images. Options:
+
+- stdout: prints all images on stdout
+- file: outputs all images to a file. (View file-name flag)
+- json: outputs all images to a file in JSON format. (View file-name flag)
+- yaml: outputs all images to a file in YAML format. (View file-name flag)
+
+Usage:
+
+	- helm-mirror inspect-images /tmp/helm --output stdout
+
+	- helm-mirror inspect-images /tmp/helm -o stdout
+
+	- helm-mirror inspect-images /tmp/helm -o file
+
+	- helm-mirror inspect-images /tmp/helm -o json
+
+	- helm-mirror inspect-images /tmp/helm -o yaml
+
+`
+
+const fileNameDesc = `cet the name of the output file.
+
+Usage:
+
+	- helm-mirror inspect-images /tmp/helm -o file --file-name images.txt
+
+	- helm-mirror inspect-images /tmp/helm -o json --file-name images.json
+
+	- helm-mirror inspect-images /tmp/helm -o yaml --file-name images.yaml
+
+`
+
+// inspectImagesCmd represents the images command
+var inspectImagesCmd = &cobra.Command{
+	Use:   "inspect-images [folder|tgzfile]",
+	Short: "Extract all the images of the Helm Charts.",
+	Long:  imagesDesc,
+	Args:  validateInspectImagesArgs,
+	RunE:  runInspectImages,
+}
+
+func init() {
+	inspectImagesCmd.PersistentFlags().StringVarP(&output, "output", "o", "stdout", outputDesc)
+	inspectImagesCmd.PersistentFlags().StringVar(&imagesFile, "file-name", "images.out", fileNameDesc)
+	rootCmd.AddCommand(inspectImagesCmd)
+}
+
+func validateInspectImagesArgs(cmd *cobra.Command, args []string) error {
+	l := utils.NewLogger(false, os.Stdout)
+	if len(args) < 1 {
+		l.Fatal("requires at least one arg to execute")
+		return errors.New("requires at least one arg")
```

Comment:
This will never happen, since `l.Fatal` will log and call `os.Exit` (pointing here but it's also used in other places). Prefer to simply return the error (log the message if you want, but I think it's a bit repetitive...), and then fallback the error until the command, in which it will simply log the message and quit (this is how usually things are done in Go).